nexus-sdk/src/main/java/io/nexusrpc/handler/ServiceHandler.java

```diff
@@ -188,7 +226,10 @@ public ServiceHandler build() {
         }
         instancesByName.put(instance.getDefinition().getName(), instance);
       }
-      return new ServiceHandler(Collections.unmodifiableMap(instancesByName), serializer);
+      return new ServiceHandler(
+          Collections.unmodifiableMap(instancesByName),
+          serializer,
+          Collections.unmodifiableList(middlewares));
```

Comment:
```suggestion
          Collections.unmodifiableList(new ArrayList<>(middlewares)));
```

Sorry, forgot to mention this before, but basically you need to copy the user-mutable collection too

Reply:
good point, done